app/actions/websocket/index.ts

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {DeviceEventEmitter} from 'react-native';
+
+import {fetchMyChannelsForTeam} from '@actions/remote/channel';
+import {fetchPostsSince} from '@actions/remote/post';
+import {loadMe, updateAllusersSinceLastDisconnect} from '@actions/remote/user';
+import {General, WebsocketEvents} from '@app/constants';
+import {SYSTEM_IDENTIFIERS} from '@app/constants/database';
+import {queryCommonSystemValues, queryWebSocketLastDisconnected} from '@app/queries/servers/system';
+import DatabaseManager from '@database/manager';
+
+import {handleLeaveTeamEvent} from './teams';
+
+export async function handleFirstConnect(serverURL: string) {
+    const database = DatabaseManager.serverDatabases[serverURL]?.database;
+    if (!database) {
+        return;
+    }
+    const {config} = await queryCommonSystemValues(database);
+    const lastDisconnect = await queryWebSocketLastDisconnected(database);
+    if (lastDisconnect && config.EnableReliableWebSockets !== 'true') {
+        doReconnect(serverURL);
+        return;
+    }
+
+    doFirstConnect(serverURL);
+}
+
+export async function handleReconnect(serverURL: string) {
+    doReconnect(serverURL);
+}
+
+export async function handleClose(serverURL: string, lastDisconnect: number) {
+    console.log('WSA: close');
+    const operator = DatabaseManager.serverDatabases[serverURL]?.operator;
+    if (!operator) {
+        return;
+    }
+    await operator.handleSystem({
+        systems: [
+            {
+                id: SYSTEM_IDENTIFIERS.WEBSOCKET,
+                value: lastDisconnect.toString(10),
+            },
+        ],
+        prepareRecordsOnly: false,
+    });
+}
+
+async function doFirstConnect(serverURL: string) {
+    console.log('WSA: doFirstConnect');
+    await updateAllusersSinceLastDisconnect(serverURL);
+}
+
+async function doReconnect(serverURL: string) {
+    console.log('WSA: doReconnect');
+
+    const database = DatabaseManager.serverDatabases[serverURL];
+    if (!database) {
+        return;
+    }
+
+    const {teamMemberships, config, error} = await loadMe(serverURL, {});
+    if (error) {
+        //TODO handle error
+        return;
+    }
+
+    const {currentUserId, currentTeamId, currentChannelId} = await queryCommonSystemValues(database.database);
+    const lastDisconnectedAt = await queryWebSocketLastDisconnected(database.database);
+    const currentTeamMembership = teamMemberships?.find((tm) => tm.team_id === currentTeamId && tm.delete_at === 0);
+
+    if (currentTeamMembership) {
+        const channelsRequest = await fetchMyChannelsForTeam(serverURL, currentTeamMembership.team_id, false, lastDisconnectedAt);
```

Comment:
This is mainly translated from V1. My guess is that "loadMe" should load all that information. Since you want to revisit the loadMe function, we can discuss this while we discuss the loadMe function.